locals.tf

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------------------------------------------------
+# Retrieve the effective Account ID, User ID, and ARN in which Terraform is
+# authorized.  This is used to calculate the session names for assumed roles.
+# ------------------------------------------------------------------------------
+data "aws_caller_identity" "default" {}
+
+# ------------------------------------------------------------------------------
+# Retrieve the information for all accounts in the organization.
+# ------------------------------------------------------------------------------
+data "aws_organizations_organization" "cool" {
+  provider = aws.organizationsreadonly
+}
+
+# ------------------------------------------------------------------------------
+# Retrieve the caller identity for the Users provider in order to get
+# the associated Account ID.
+# ------------------------------------------------------------------------------
+data "aws_caller_identity" "users" {
+  provider = aws.users
+}
+
+# ------------------------------------------------------------------------------
+# Evaluate expressions for use throughout this configuration.
+# ------------------------------------------------------------------------------
+locals {
+  # Extract the user name of the current caller for use
+  # as assume role session names.
+  caller_user_name = split("/", data.aws_caller_identity.default.arn)[1]
+
+  # Get IDs of all non-assessment accounts in the organization, i.e. those
+  # that don't have account names like: "env[:digit:] (.*)"
+  all_non_assessment_account_ids = [
+    for account in data.aws_organizations_organization.cool.accounts :
+    account.id
+    if length(regexall("^env[[:digit:]]+ \\(.*\\)$", account.name)) == 0
+  ]
+
+  # Create a list of all provision roles in non-assessment accounts.
+  all_non_assessment_provision_roles = formatlist("arn:aws:iam::%s:role/%s", local.all_non_assessment_account_ids, var.provision_assessment_role_name)
+
+  # Assumption of the following non-assessment account roles is required
+  # to successfully provision assessment environments.
+  # TODO: Determine if it is possible/worthwhile to replace any
+  # "provision account" roles below with something less powerful.  New roles
+  # would need to be created in appropriate repositories, then used in
+  # cisagov/cool-assessment-terraform and also included below.
+  # See https://github.com/cisagov/cool-assessment-terraform/issues/133.
+  required_non_assessment_roles = [
```

Comment:
The only roles that can end up in `local.all_non_assessment_provision_roles` are those that are named the same as `var.provision_assessment_role_name` (which defaults to `ProvisionAccount`).  

However you made me realize that `required_non_assessment_roles ` should indeed include `data.terraform_remote_state.master.outputs.organizationsreadonly_role.arn`, which led to https://github.com/cisagov/cool-assessment-provisioner-iam/pull/1/commits/6f16044fe41c79ebbbf9ec2feed69038f8d3beb1 and https://github.com/cisagov/cool-assessment-provisioner-iam/pull/1/commits/6f04d3545e598639ce6f9380cce59d290132dece, so thanks for mentioning this!

Reply:
Ah, right. Sorry I glossed over that role name restriction because of the other ARNs in `required_non_assessment_roles` (the ones that don't look like `provisionaccount_role`). The core question of "should this be whitelisted" is answered however so I'm glad I asked.

Reply:
> The core question of "should this be whitelisted" is answered however so I'm glad I asked.

So am I! 👍